src/main/osd/osd.c

```diff
@@ -413,6 +418,12 @@ static void osdCompleteInitialization(void)
     osdAnalyzeActiveElements();
     displayCommitTransaction(osdDisplayPort);
 
+#if defined(USE_MAX7456) || defined(USE_FRSKYOSD)
+    task_t *taskOsd;
+    taskOsd = getTask(TASK_OSD);
+    taskOsd->desiredPeriodUs = TASK_PERIOD_HZ(osdConfig()->task_frequency);
+#endif
```

Comment:
This logic should not be here. It should be in `tasksInit()` in `fc/tasks.c` before the task is enabled here: https://github.com/betaflight/betaflight/blob/5f8d9da384fb84fb3a1d56a1ede9510f4c2d712b/src/main/fc/tasks.c#L337-L339

Also there should not be any dependency on the OSD device type.